tests/conftest.py

```diff
@@ -0,0 +1,167 @@
+
+
+@pytest.fixture(scope="session")
+def service_url(environment):
+    base_url = "https://api.service.nhs.uk" if environment == "prod" else f"https://{environment}.api.service.nhs.uk"
+    service_base_path = get_env('SERVICE_BASE_PATH')
+
+    return f"{base_url}/{service_base_path}"
+
+
+@pytest.fixture(scope="session")
+def status_endpoint_api_key():
+    return get_env("STATUS_ENDPOINT_API_KEY")
+
+
+@pytest.fixture(scope="session")
+def id_token_key():
+    id_token_key_path = get_env("ID_TOKEN_TESTING_PRIVATE_KEY_ABSOLUTE_PATH")
+
+    with open(id_token_key_path, "r") as f:
+        return f.read()
+
+
+@pytest.fixture(scope="session")
+def asid():
+    return DEFAULT_ASID
+
+
+@pytest.fixture(scope="session")
+def user_id():
+    return DEFAULT_USERID
+
+
+@pytest.fixture
+async def product(environment, service_name):
+    # Setup
+    product = ApigeeApiProducts()
+    await product.create_new_product()
+
+    print('CREATED PRODUCT NAME: ' + product.name)
+
+    # Update products allowed paths
+    await product.update_proxies([service_name, 'identity-service-' + environment])
+
+    await product.update_scopes(
+        scopes=["urn:nhsd:apim:user-nhs-id:aal3:e-referrals-service-api"]
+    )
+    yield product
+
+    # Teardown
+    print('Cleanup product')
+    await product.destroy_product()
+
+
+@pytest.fixture
+async def app(environment, product, asid):
+    # Setup
+    app = ApigeeApiDeveloperApps()
+    await app.create_new_app()
+
+    print('CREATED APP NAME: ' + app.name)
+
+    # Set default JWT Testing resource url and ASID
+    await app.set_custom_attributes({
+            'asid': asid,
+            'jwks-resource-url': 'https://raw.githubusercontent.com/NHSDigital/'
+                                 f'identity-service-jwks/main/jwks/{environment}/'
+                                 '9baed6f4-1361-4a8e-8531-1f8426e3aba8.json'
+    })
+
+    # Assign the new app to the product
+    await app.add_api_product([product.name])
+
+    yield app
+
+    # Teardown
+    print('Cleanup app')
+    await app.destroy_app()
+
+
+@pytest.fixture
+async def access_code(app, id_token_key, user_id):
+    oauth = OauthHelper(app.client_id, app.client_secret, app.callback_url)
+
+    claims = {
+            'at_hash': 'tf_-lqpq36lwO7WmSBIJ6Q',
+            'sub': user_id,
+            'auditTrackingId': '91f694e6-3749-42fd-90b0-c3134b0d98f6-1546391',
+            'amr': ['N3_SMARTCARD'],
+            'iss': 'https://am.nhsint.auth-ptl.cis2.spineservices.nhs.uk:443/'
+                   'openam/oauth2/realms/root/realms/NHSIdentity/realms/Healthcare',
+            'tokenName': 'id_token',
+            'aud': '969567331415.apps.national',
+            'c_hash': 'bc7zzGkClC3MEiFQ3YhPKg',
+            'acr': 'AAL3_ANY',
+            'org.forgerock.openidconnect.ops': '-I45NjmMDdMa-aNF2sr9hC7qEGQ',
+            's_hash': 'LPJNul-wow4m6Dsqxbning',
+            'azp': '969567331415.apps.national',
+            'auth_time': 1610559802,
```

Comment:
This is the time the authentication occurred. So I guess it is just now. Probably `int(time())` like below for exp (without the + 6000).

Reply:
I've now pushed this change.